src/test/ui/fn_once-moved.stderr

```diff
@@ -0,0 +1,13 @@
+error[E0382]: use of moved value: `debug_dump_dict`
+  --> $DIR/fn_once-moved.rs:21:5
+   |
+20 |     debug_dump_dict();
+   |     --------------- value moved here
+21 |     debug_dump_dict();
+   |     ^^^^^^^^^^^^^^^ value used here after move
+   |
+   = help: closure was moved because it only implements `FnOnce`
+   = note: move occurs because `debug_dump_dict` has type `[closure@$DIR/fn_once-moved.rs:15:27: 19:6 dict:std::collections::HashMap<i32, i32>]`, which does not implement the `Copy` trait
```

Comment:
Personally, I think it should be removed. I highly doubt anyone will extract much value out of it -- particularly in non-trivial cases with more upvars. It'd be better to extend the upvar inference to track why a move is occurring, I suspect. 

@GuillaumeGomez, interested in pursuing that?